docs/index.md

```diff
@@ -60,34 +65,158 @@ resource "ocm_group_membership" "my_admin" {
 
 The value of the `token` attribute of the provider should be the OCM
 authentication token that you can get [here](https://console.redhat.com/openshift/token).
-If this attribute isn't used then the provider will try to get the token from
+If this attribute isn't used then the provider will try to get the token it from
 the `OCM_TOKEN` environment variable.
 
-## Schema
+### Create AWS Rosa STS Cluster
+
+The following example shows a production grade rosa cluster with:
+
+* Existing VPC & Subnets
+* Multi AZ
+* Proxy
+* STS
+
+```
+data "aws_caller_identity" "current" {}
+
+locals {
+  sts_roles = {
+      role_arn = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/ManagedOpenShift-Installer-Role",
+      support_role_arn = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/ManagedOpenShift-Support-Role",
+      operator_iam_roles = [
+        {
+          name =  "cloud-credential-operator-iam-ro-creds",
+          namespace = "openshift-cloud-credential-operator",
+          role_arn = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/${var.operator_role_prefix}-openshift-cloud-credential-operator-cloud-c",
+        },
+        {
+          name =  "installer-cloud-credentials",
+          namespace = "openshift-image-registry",
+          role_arn = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/${var.operator_role_prefix}-openshift-image-registry-installer-cloud-cr",
+        },
+        {
+          name =  "cloud-credentials",
+          namespace = "openshift-ingress-operator",
+          role_arn = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/${var.operator_role_prefix}-openshift-ingress-operator-cloud-credential",
+        },
+        {
+          name =  "ebs-cloud-credentials",
+          namespace = "openshift-cluster-csi-drivers",
+          role_arn = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/${var.operator_role_prefix}-openshift-cluster-csi-drivers-ebs-cloud-cre",
+        },
+        {
+          name =  "cloud-credentials",
+          namespace = "openshift-cloud-network-config-controller",
+          role_arn = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/${var.operator_role_prefix}-openshift-cloud-network-config-controller-c",
+        },
+        {
+          name =  "aws-cloud-credentials",
+          namespace = "openshift-machine-api",
+          role_arn = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/${var.operator_role_prefix}-openshift-machine-api-aws-cloud-credentials",
+        },
+      ]
+      instance_iam_roles = {
+        master_role_arn = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/ManagedOpenShift-ControlPlane-Role",
+        worker_role_arn = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/ManagedOpenShift-Worker-Role"
+      },    
+  }
+}
+resource "ocm_cluster" "rosa_cluster" {
```

Comment:
Please use the dedicated resource ocm_cluster_rosa_classic
ocm_cluster doesn't support STS anymore... 
Update it like in that example -https://github.com/terraform-redhat/terraform-provider-ocm/blob/main/examples/create_rosa_cluster/create_rosa_sts_cluster/cluster/main.tf#L50